test/bdd/features/introduce.feature

```diff
@@ -54,6 +54,18 @@ Feature: Introduce protocol
     Then  "Carol" has did exchange connection with "Bob"
     And   "Carol" checks the history of introduce protocol events "deciding,deciding,waiting,waiting,done,done" and stop
 
+  @proposal_response_with_embedded_route_request
+  Scenario: Bob sends a response with approve and an out-of-band request with an embedded route-request.
+    Given   "Alice-Router,Bob" exchange DIDs with "Alice"
+    When   "Alice" sends introduce proposal to the "Alice-Router" and "Bob"
+    And   "Alice-Router" wants to know "Bob" and sends introduce response with approve and provides an out-of-band request with an embedded "route-request"
+    And   "Bob" wants to know "Alice-Router" and sends introduce response with approve
+    Then   "Alice" checks the history of introduce protocol events "arranging,arranging,arranging,arranging,arranging,arranging,arranging,arranging,delivering,delivering,confirming,confirming,done,done"
```

Comment:
it takes a lot of arranging ...

Reply:
Agreed. I reused the existing infra in the `introduce` bdd test package after realizing that approach involves less work than implementing a terse BDD test...

Reply:
I think we can rewrite this logic, instead of checking each event, we can check whether the consistency is ok. e.g `arranging,done` or `arranging,delivering,done`. I did it to be sure that all events were executed as I expected.